runtime/lua/vim/lsp/util.lua

```diff
@@ -0,0 +1,115 @@
+local util = {
+  ui = require('vim.lsp.util.ui'),
+}
+
+local get_buffer_lines = function(bufnr)
+  return vim.api.nvim_buf_get_lines(bufnr, 0, -1, false)
+end
+
+util.get_buffer_text = function(bufnr)
+  return table.concat(get_buffer_lines(bufnr), '\n')
+end
+
+util.get_filetype = function(bufnr)
+  bufnr = bufnr or vim.api.nvim_get_current_buf()
+  return vim.api.nvim_buf_get_option(bufnr, 'filetype')
+end
+
+util.decode_json = function(data)
+  return vim.api.nvim_call_function('json_decode', {data})
```

Comment:
Since this is going to be a hot function, it might be worth it to use `cjson` here for encode/decode.

Reply:
> Since this is going to be a hot function, it might be worth it to use `cjson` here for encode/decode.

Well, I think we should better keep external dependencies to the minimum, shoudln't we?

Reply:
Later on we should implement direct access to our json decoder from lua without viml overhead. Let's keep this for now until then.